[Bug] When cleaning up a manifest, the name of the manifest is null

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.


### Paimon version

0.8 snapshot

### Compute Engine

flink 1.17.2

### Minimal reproduce step

changelog-producer=none
![image](https://github.com/apache/paimon/assets/40812600/84e3feb4-d80a-49e2-8fac-914370a35f63)
![image](https://github.com/apache/paimon/assets/40812600/6d17a9bf-475c-423d-b1e0-e6bd48894119)


### What doesn't meet your expectations?

When cleaning up a manifest, the name of the manifest is null

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

Caused by: java.io.FileNotFoundException: File 'hdfs://nameHAservice/user/hive/warehouse/xqy_portal_nifitest.db/paimon_ts_pftccore_pftc_rs_status/manifest/null' not found, Possible causes: 1.snapshot expires too fast, you can configure 'snapshot.time-retained' option with a larger value. 2.consumption is too slow, you can improve the performance of consumption (For example, increasing parallelism).
	at org.apache.paimon.utils.FileUtils.checkExists(FileUtils.java:124) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.FileUtils.createFormatReader(FileUtils.java:143) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.readWithIOException(ObjectsFile.java:125) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:108) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	... 30 more
2024-04-08 09:01:58,433 WARN  org.apache.paimon.operation.FileDeletionBase                 [] - Failed to read manifest list file null
java.lang.RuntimeException: Failed to read manifest list null
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:110) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:82) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:78) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.tryReadManifestList(FileDeletionBase.java:215) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.cleanUnusedManifestList(FileDeletionBase.java:180) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.cleanUnusedManifests(FileDeletionBase.java:201) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.ExpireSnapshotsImpl.expireUntil(ExpireSnapshotsImpl.java:225) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.ExpireSnapshotsImpl.expire(ExpireSnapshotsImpl.java:135) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.AbstractFileStoreTable.lambda$newCommit$2(AbstractFileStoreTable.java:332) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expireSnapshots(TableCommitImpl.java:359) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:346) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.lambda$expire$9(TableCommitImpl.java:332) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:321) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:329) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.commitMultiple(TableCommitImpl.java:205) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.StoreCommitter.commit(StoreCommitter.java:100) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.StoreMultiCommitter.commit(StoreMultiCommitter.java:142) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.CommitterOperator.commitUpToCheckpoint(CommitterOperator.java:160) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.CommitterOperator.endInput(CommitterOperator.java:147) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.endOperatorInput(StreamOperatorWrapper.java:96) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.RegularOperatorChain.endInput(RegularOperatorChain.java:97) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.io.StreamOneInputProcessor.processInput(StreamOneInputProcessor.java:68) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.processInput(StreamTask.java:550) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:231) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:839) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:788) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:952) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:931) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:745) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:562) [flink-dist-1.17.2.jar:1.17.2]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.io.FileNotFoundException: File 'hdfs://nameHAservice/user/hive/warehouse/xqy_portal_nifitest.db/paimon_ts_pftccore_pftc_rs_status/manifest/null' not found, Possible causes: 1.snapshot expires too fast, you can configure 'snapshot.time-retained' option with a larger value. 2.consumption is too slow, you can improve the performance of consumption (For example, increasing parallelism).
	at org.apache.paimon.utils.FileUtils.checkExists(FileUtils.java:124) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.FileUtils.createFormatReader(FileUtils.java:143) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.readWithIOException(ObjectsFile.java:125) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:108) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	... 30 more
2024-04-08 09:01:58,517 WARN  org.apache.paimon.operation.FileDeletionBase                 [] - Failed to read manifest list file null
java.lang.RuntimeException: Failed to read manifest list null
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:110) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:82) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:78) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.tryReadManifestList(FileDeletionBase.java:215) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.cleanUnusedManifestList(FileDeletionBase.java:180) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.cleanUnusedManifests(FileDeletionBase.java:201) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.ExpireSnapshotsImpl.expireUntil(ExpireSnapshotsImpl.java:225) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.ExpireSnapshotsImpl.expire(ExpireSnapshotsImpl.java:135) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.AbstractFileStoreTable.lambda$newCommit$2(AbstractFileStoreTable.java:332) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expireSnapshots(TableCommitImpl.java:359) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:346) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.lambda$expire$9(TableCommitImpl.java:332) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:321) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:329) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.commitMultiple(TableCommitImpl.java:205) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.StoreCommitter.commit(StoreCommitter.java:100) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.StoreMultiCommitter.commit(StoreMultiCommitter.java:142) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.CommitterOperator.commitUpToCheckpoint(CommitterOperator.java:160) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.CommitterOperator.endInput(CommitterOperator.java:147) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.endOperatorInput(StreamOperatorWrapper.java:96) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.RegularOperatorChain.endInput(RegularOperatorChain.java:97) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.io.StreamOneInputProcessor.processInput(StreamOneInputProcessor.java:68) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.processInput(StreamTask.java:550) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:231) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:839) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:788) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:952) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:931) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:745) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:562) [flink-dist-1.17.2.jar:1.17.2]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.io.FileNotFoundException: File 'hdfs://nameHAservice/user/hive/warehouse/xqy_portal_nifitest.db/paimon_ts_pftccore_pftc_rs_status/manifest/null' not found, Possible causes: 1.snapshot expires too fast, you can configure 'snapshot.time-retained' option with a larger value. 2.consumption is too slow, you can improve the performance of consumption (For example, increasing parallelism).
	at org.apache.paimon.utils.FileUtils.checkExists(FileUtils.java:124) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.FileUtils.createFormatReader(FileUtils.java:143) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.readWithIOException(ObjectsFile.java:125) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:108) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	... 30 more
2024-04-08 09:01:58,734 WARN  org.apache.paimon.operation.FileDeletionBase                 [] - Failed to read manifest list file null
java.lang.RuntimeException: Failed to read manifest list null
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:110) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:82) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:78) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.tryReadManifestList(FileDeletionBase.java:215) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.cleanUnusedManifestList(FileDeletionBase.java:180) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.operation.FileDeletionBase.cleanUnusedManifests(FileDeletionBase.java:201) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.ExpireSnapshotsImpl.expireUntil(ExpireSnapshotsImpl.java:225) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.ExpireSnapshotsImpl.expire(ExpireSnapshotsImpl.java:135) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.AbstractFileStoreTable.lambda$newCommit$2(AbstractFileStoreTable.java:332) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expireSnapshots(TableCommitImpl.java:359) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:346) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.lambda$expire$9(TableCommitImpl.java:332) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.shade.guava30.com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:321) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:329) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.table.sink.TableCommitImpl.commitMultiple(TableCommitImpl.java:205) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.StoreCommitter.commit(StoreCommitter.java:100) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.StoreMultiCommitter.commit(StoreMultiCommitter.java:142) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.CommitterOperator.commitUpToCheckpoint(CommitterOperator.java:160) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.flink.sink.CommitterOperator.endInput(CommitterOperator.java:147) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.flink.streaming.runtime.tasks.StreamOperatorWrapper.endOperatorInput(StreamOperatorWrapper.java:96) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.RegularOperatorChain.endInput(RegularOperatorChain.java:97) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.io.StreamOneInputProcessor.processInput(StreamOneInputProcessor.java:68) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.processInput(StreamTask.java:550) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:231) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:839) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:788) ~[flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:952) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:931) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:745) [flink-dist-1.17.2.jar:1.17.2]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:562) [flink-dist-1.17.2.jar:1.17.2]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.io.FileNotFoundException: File 'hdfs://nameHAservice/user/hive/warehouse/xqy_portal_nifitest.db/paimon_ts_pftccore_pftc_rs_status/manifest/null' not found, Possible causes: 1.snapshot expires too fast, you can configure 'snapshot.time-retained' option with a larger value. 2.consumption is too slow, you can improve the performance of consumption (For example, increasing parallelism).
	at org.apache.paimon.utils.FileUtils.checkExists(FileUtils.java:124) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.FileUtils.createFormatReader(FileUtils.java:143) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.readWithIOException(ObjectsFile.java:125) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	at org.apache.paimon.utils.ObjectsFile.read(ObjectsFile.java:108) ~[paimon-flink-1.17-0.8-SNAPSHOT.jar:0.8-SNAPSHOT]
	... 30 more

--- a/paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java
+++ b/paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java
@@ -197,7 +197,7 @@ public abstract class FileDeletionBase {
             Snapshot snapshot, Set<String> skippingSet, boolean deleteChangelog) {
         cleanUnusedManifestList(snapshot.baseManifestList(), skippingSet);
         cleanUnusedManifestList(snapshot.deltaManifestList(), skippingSet);
-        if (deleteChangelog) {
+        if (deleteChangelog && snapshot.changelogManifestList() != null) {
             cleanUnusedManifestList(snapshot.changelogManifestList(), skippingSet);
         }
         cleanUnusedIndexManifests(snapshot, skippingSet);

